### PR TITLE
parser: support cast as double

### PR DESCRIPTION
**Pull request by wshwsh12**
### What problem does this PR solve? 

Support `cast(xxx as double)`.
Mysql support the parser in 8.0.17.
https://dev.mysql.com/doc/refman/8.0/en/cast-functions.html

### CODE REVIEW

**Comment by winoros** [comment]
This pr only updates the parser?

**Comment by wshwsh12** [comment]
> This pr only updates the parser?

Yes

File: expression/builtin_cast.go [hunk 1]
```diff
@@ -458,7 +458,7 @@ func (b *builtinCastIntAsRealSig) evalReal(row chunk.Row) (res float64, isNull b
 	if isNull || err != nil {
 		return res, isNull, err
 	}
-	if !mysql.HasUnsignedFlag(b.tp.Flag) {
+	if !mysql.HasUnsignedFlag(b.tp.Flag) && !mysql.HasUnsignedFlag(b.args[0].GetType().Flag) {
```
Comment (XuHuaiyu) [inline expression/builtin_cast.go]: We need to add a test case to cover cast(unsigned_int as double)

Comment (wshwsh12) [reply expression/builtin_cast.go]: update.

**Comment by zz-jason** [comment]
@wshwsh12 Please update the document as well.

**Comment by wshwsh12** [comment]
> @wshwsh12 Please update the document as well.

OK

**Pull request merged**